packages/browser/lib/Browser.ts

```diff
@@ -0,0 +1,94 @@
+/// <reference types="node" />
+import { Client, Event, IAdapter, IBreadcrumb, IUser } from '@sentry/core';
+const Raven = require('raven-js');
+
+export interface IBrowserOptions {
+  allowSecretKey?: boolean;
+  allowDuplicates?: boolean;
+}
+
+export class Browser implements IAdapter {
+  private client: Client;
+
+  constructor(client: Client, public options: IBrowserOptions = {}) {
+    this.client = client;
+    return this;
+  }
+
+  public install(): Promise<boolean> {
+    Raven.config(this.client.dsn.getDsn(false), this.options).install();
+    return Promise.resolve(true);
+  }
+
+  public getRaven() {
+    return Raven;
+  }
+
+  public setOptions(options: IBrowserOptions) {
+    Object.assign(this.options, options);
+    Object.assign(Raven._globalOptions, this.options);
+    return this;
+  }
+
+  public captureException(exception: Error) {
+    const ravenSendRequest = Raven._sendProcessedPayload;
```

Comment:
Event though the promise executor function is invoked immediately, I'd move this line (storing the original `sendProcessedPayload` inside of it to avoid the looks of a possible race condition. And then also immediately map it back in the callback **before** calling `resolve`. 
Same goes for the other methods with the same pattern.